observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,221 @@
+    this.perMethodFilters = ImmutableMap.copyOf(perMethodFilters);
+    if (!perServiceFilters.isEmpty() || !perMethodFilters.isEmpty()) {
+      this.methodOrServiceFilterPresent = true;
+    }
+  }
+
+  @VisibleForTesting
+  void setEventFilterSet() {
+    List<EventType> eventFilters = config.getEventTypes();
+    if (eventFilters == null) {
+      return;
+    }
+    if (eventFilters.isEmpty()) {
+      this.logEventTypeSet = ImmutableSet.of();
+      return;
+    }
+    this.logEventTypeSet = ImmutableSet.copyOf(eventFilters);
+  }
+
+  /**
+   * Class containing results for method/service filter information, such as flag for logging
+   * method/service and payload limits to be used for filtering.
+   */
+  @AutoValue
+  public abstract static class FilterParams {
+
+    abstract boolean log();
+
+    abstract int headerBytes();
+
+    abstract int messageBytes();
+
+    @VisibleForTesting
+    public static FilterParams create(boolean log, int headerBytes, int messageBytes) {
+      return new AutoValue_ConfigFilterHelper_FilterParams(
+          log, headerBytes, messageBytes);
+    }
+  }
+
+  /**
+   * Checks if the corresponding service/method passed needs to be logged as per the user provided
+   * configuration.
+   *
+   * @param method the fully qualified name of the method
+   * @return MethodFilterParams object 1. specifies if the corresponding method needs to be logged
+   *     (log field will be set to true) 2. values of payload limits retrieved from configuration
+   */
+  public FilterParams isMethodToBeLogged(MethodDescriptor<?, ?> method) {
+    FilterParams params = NO_FILTER_PARAMS;
+    if (methodOrServiceFilterPresent) {
+      String fullMethodName = method.getFullMethodName();
+      if (perMethodFilters.containsKey(fullMethodName)) {
+        params = perMethodFilters.get(fullMethodName);
+      } else {
+        String serviceName = method.getServiceName();
+        if (perServiceFilters.containsKey(serviceName)) {
+          params = perServiceFilters.get(serviceName);
+        } else if (perServiceFilters.containsKey(globalPattern)) {
+          params = perServiceFilters.get(globalPattern);
+        }
+      }
+    }
+    return params;
+  }
+
+  /**
+   * Checks if the corresponding event passed needs to be logged as per the user provided
+   * configuration.
+   *
+   * <p> All events are logged by default if event_types is not specified or {} in configuration.
+   * If event_types is specified as [], no events will be logged.
+   * If events types is specified as a non-empty list, only the events specified in the
+   * list will be logged.
+   * </p>
+   *
+   * @param event gRPC observability event
+   * @return true if event needs to be logged, false otherwise
+   */
+  public boolean isEventToBeLogged(EventType event) {
+    if (logEventTypeSet == null) {
+      return true;
+    }
+    boolean logEvent;
+    if (logEventTypeSet.isEmpty()) {
+      logEvent = false;
```

Comment:
if `logEventTypeSet.isEmpty()` then `logEventTypeSet.contains(event)` anyway returns `false` so I don't understand the need for this special case code.